common/src/main/java/com/tw/go/plugin/BuildStatusNotifierPlugin.java

```diff
@@ -154,21 +154,21 @@ GoPluginApiResponse handleStageNotification(GoPluginApiRequest goPluginApiReques
             List<Map> materialRevisions = (List<Map>) pipeline.get("build-cause");
             for (Map materialRevision : materialRevisions) {
                 Map material = (Map) materialRevision.get("material");
-                if (isMaterialOfType(material, provider.pollerPluginId())) {
+                if (isMaterialFromTypes(material, provider.pollerPluginIds())) {
                     Map materialConfiguration = (Map) material.get("scm-configuration");
                     String url = (String) materialConfiguration.get("url");
 
                     List<Map> modifications = (List<Map>) materialRevision.get("modifications");
                     String revision = (String) modifications.get(0).get("revision");
                     Map modificationData = (Map) modifications.get(0).get("data");
-                    String prId = (String) modificationData.get("PR_ID");
+                    String prBranch = (String) modificationData.getOrDefault("PR_BRANCH", "PR_ID");
```

Comment:
Realised we don't use this field of updateStatus for any other provider except for Gitlab. Earlier implementation was pushing the commit status as part of the merge-request refspec instead of the source branch. That's the reason why the commit status wouldn't show up on the Merge Request page.